pkg/ddl/index.go

```diff
@@ -1271,14 +1281,156 @@ func initForReorgIndexes(w *worker, job *model.Job, idxInfos []*model.IndexInfo)
 	return nil
 }
 
-func (w *worker) analyzeTableAfterCreateIndex(job *model.Job, dbName, tblName string) bool {
+type analyzeStatus int
+
+const (
+	analyzeUnknown analyzeStatus = iota
+	analyzeRunning
+	analyzeFinished
+	analyzeFailed
+)
+
+// queryAnalyzeStatusSince queries `SHOW ANALYZE STATUS` for the specified table
+// and start_time >= startTS. It returns one of analyzeStatus values. If the
+// sessPool cannot provide a session or the query fails, it returns
+// analyzeUnknown and a nil error so the caller may decide to proceed with
+// starting ANALYZE.
+func (w *worker) queryAnalyzeStatusSince(startTS uint64, dbName, tblName string) (analyzeStatus, error) {
+	sessCtx, err := w.sessPool.Get()
+	if err != nil {
+		return analyzeUnknown, err
+	}
+	defer w.sessPool.Put(sessCtx)
+
+	// internal sql may not init the analysis related variable correctly.
+	restore, err := statsutil.UpdateSCtxVarsForStats(sessCtx)
```

Comment:
Even if we want to handle the timezone issue here, I don’t think it’s reasonable to put it in this utility function, since it’s only meant for analyze-related variables. This seems to be just a special case specific to this function.